torch/csrc/Size.cpp

```diff
@@ -23,10 +23,19 @@ static PyObject * THPSize_pynew(PyTypeObject *type, PyObject *args, PyObject *kw
   if (self) {
     for (Py_ssize_t i = 0; i < PyTuple_Size(self); ++i) {
       PyObject *item = PyTuple_GET_ITEM(self.get(), i);
-      if (!THPUtils_checkLong(item)) {
-        return PyErr_Format(PyExc_TypeError, "torch.Size() takes an iterable of 'int' (item %zd is '%s')",
-            i, Py_TYPE(item)->tp_name);
+      if (THPUtils_checkLong(item)) {
+        continue;
       }
+      // item.__index__() works with 0-dim tensors and tensors with one element
+      THPObjectPtr number(PyNumber_Index(item));
+      if (number && THPUtils_checkLong(number.get())) {
```

Comment:
unpackIndex here.

Reply:
unpackIndex returns an int64_t. Here I wanted to return a "python int". The logic is that I'm replacing the elements (including any scalar tensors) of the tuple that the user has passed to torch.Size with "python ints".